weave/index/index_sqlite.py

```diff
@@ -366,6 +370,11 @@ def get_parents(self, basket_address, **kwargs):
                     )
 
         parent_df.drop(columns="path", inplace=True)
+
+        # Keep all of the rows where the generation_level is less than or
+        # equal to the max_gen_level
+        parent_df = parent_df[parent_df["generation_level"] <= max_gen_level]
```

Comment:
Same as above, let's change the SQL command itself instead of filtering after the fact.